### DISCUSSION TITLE
data.db文件过大

### DISCUSSION THREAD

**Discussion by NoEggEgg** [AlistGo/alist#1401] [Q&A]
目前data.db文件达到了2.51GB，应该是表x_search_files导致的，网页端是否可以增加清除按钮，目前应该如何处理

**Comment by blacklin** [comment]
用sqlite3打开data.db，执行 `DELETE FROM x_search_files; ` 试试

**Comment by anwen-anyi** [comment]
`x_search_files`数据库表 有缓存应该是你开启搜索`(enable search)`导致的~
删除的话可以使用楼上的老哥方案试试，搜索功能处于实验室功能,还有就是你不用搜索功能建议关闭...QAQ
楼上老哥方案不行的话可以用`navicat`来链接打开然后选择删除数据也可以的.

**Reply by NoEggEgg** [reply]
一般电脑都不会装操作数据库的软件，我去试试，建议可以在网页端增加清除功能，方便一些。

**Reply by KinShin** [reply]
删除了相应的表也不会减少占用大小，因为 sqlite3 默认的 auto_vacuum 参数是 NONE。

> The default setting for auto-vacuum is 0 or "none", unless the [SQLITE_DEFAULT_AUTOVACUUM](https://www.sqlite.org/compile.html#default_autovacuum) compile-time option is used. The "none" setting means that auto-vacuum is disabled. When auto-vacuum is disabled and data is deleted data from a database, the database file remains the same size. Unused database file pages are added to a "[freelist](https://www.sqlite.org/fileformat2.html#freelist)" and reused for subsequent inserts. So no database file space is lost. However, the database file does not shrink. In this mode the [VACUUM](https://www.sqlite.org/lang_vacuum.html) command can be used to rebuild the entire database file and thus reclaim unused disk space.

解决方法是手动执行 VACUUM 命令。
